compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmPackageGen.java

```diff
@@ -131,14 +130,14 @@
     private final Map<String, BIRFunctionWrapper> birFunctionMap;
     private final Map<String, String> externClassMap;
     private final Map<String, String> globalVarClassMap;
-    private final Map<String, PackageID> dependentModules;
+    private final Set<PackageID> dependentModules;
     private final BLangDiagnosticLog dlog;
 
     JvmPackageGen(SymbolTable symbolTable, PackageCache packageCache, BLangDiagnosticLog dlog) {
         birFunctionMap = new HashMap<>();
         globalVarClassMap = new HashMap<>();
         externClassMap = new HashMap<>();
-        dependentModules = new LinkedHashMap<>();
+        dependentModules = new LinkedHashSet<>();
```

Comment:
Shall we add the perf numbers before and after the fix?

Reply:
I did the change based on the flame graph. The `getPackageName` consumed most of the CPU in the  `generateDependencyList` method. By this change we remove this call, the cpu consumption of `generateDependencyList` was dropped to that was before the encoding feature was introduced.